test/src/test/java/org/corfudb/integration/CorfuReplicationLargeTxIT.java

```diff
@@ -0,0 +1,324 @@
+        // tables
+        int totalStreamingUpdates =
+            expectedStreamingUpdatesPerTable * mapNameToMapStandby.size();
+
+        CountDownLatch streamingUpdatesLatch =
+            new CountDownLatch(totalStreamingUpdates);
+        StreamingUpdateListener streamingUpdateListener =
+            new StreamingUpdateListener(streamingUpdatesLatch);
+        corfuStoreStandby.subscribeListener(streamingUpdateListener,
+            DefaultLogReplicationConfigAdapter.NAMESPACE,
+            DefaultLogReplicationConfigAdapter.TAG_ONE);
+
+        startLogReplicatorServersWithCustomMaxWriteSize(maxWriteSize);
+
+        log.debug("Wait for snapshot sync to finish");
+        statusUpdateLatch.await();
+        streamingUpdatesLatch.await();
+
+        // Verify that updates were received for all replicated tables with data
+        Assert.assertEquals(mapNameToMapStandby.size(),
+            streamingUpdateListener.getTableNameToUpdatesMap().size());
+
+        mapNameToMapStandby.keySet().forEach(key ->
+            Assert.assertTrue(streamingUpdateListener.getTableNameToUpdatesMap()
+                .containsKey(key)));
+
+
+        // Verify that the right number of entries are contained in the
+        // streaming update.  Also verify that the first entry is a 'clear'
+        // followed by all updates.
+        for (List<List<CorfuStreamEntry<Sample.StringKey,
+            SampleSchema.ValueFieldTagOne, Sample.Metadata>>> outerList :
+            streamingUpdateListener.getTableNameToUpdatesMap().values()) {
+
+            Assert.assertEquals(expectedStreamingUpdatesPerTable, outerList.size());
+
+            int totalEntriesReceived = 0;
+            List<CorfuStreamEntry<Sample.StringKey,
+                SampleSchema.ValueFieldTagOne, Sample.Metadata>> updateEntries =
+                new ArrayList<>();
+
+            for (int i=0; i<outerList.size(); i++) {
+                // If the snapshot apply was chunked, the first entry in the
+                // first chunk must be a 'clear'
+                if (i == 0) {
+                    Assert.assertEquals(CorfuStreamEntry.OperationType.CLEAR,
+                        outerList.get(i).get(0).getOperation());
+                    for (int j = 1; j<outerList.get(i).size(); j++) {
+                        updateEntries.add(outerList.get(i).get(j));
+                    }
+                } else {
+                  updateEntries.addAll(outerList.get(i));
+                }
+                totalEntriesReceived += outerList.get(i).size();
+            }
+            // The total entries received must be equal to (numUpdates + clear)
+            Assert.assertEquals(numEntriesToWrite+1, totalEntriesReceived);
+
+            updateEntries.forEach(entry -> Assert.assertEquals(
+                CorfuStreamEntry.OperationType.UPDATE, entry.getOperation()));
+        }
+
+        verifyDataOnReceiver(numEntriesToWrite);
+
+        executorService.shutdownNow();
```

Comment:
To make methods a bit smaller you can write a shutdown method and place all this code.

Reply:
moved the common shutdown code.  Kept unsubscribing of listeners in the current method as it can change with tests.